### ISSUE TITLE
feat(op-reth): [User API enhancements] Ability to forward transactions to the sequencer

### ISSUE THREAD

**Issue by merklefruit** [paradigmxyz/reth#3797]
### Describe the feature

**Diff Category**: User API enhancements | API Backend

### Overview

In Optimism, the sequencer is responsible for transaction ordering and block building. As such, transactions must be forwarded to it in order to land onchain. In this case, the node acts as a proxy between the user and the sequencer.

Blocked by:

- #3794

### Where should this feature live?

This feature will probably need to be added to Reth's RPC API in the `rpc` crate. 
For instance, here's where the `eth_sendRawTransaction` method is implemented:

https://github.com/paradigmxyz/reth/blob/90b6d006d1fc660c4883a4335e42f8e760c4adbe/crates/rpc/rpc/src/eth/api/transactions.rs#L381

### Completion Criteria

- [ ] forward transaction requests to the sequencer endpoint if the corresponding CLI flag is configured

### Additional context

_Resources_

- [`op-geth` diff](https://op-geth.optimism.io/)
- [`op-erigon` diff](https://op-erigon.testinprod.io/)
- [Optimism Specs](https://github.com/ethereum-optimism/optimism/tree/develop/specs)

**Comment by cjeva10** [comment]
We'd want to split this out more I assume, but just a first draft:

```rust
    async fn send_raw_transaction(&self, tx: Bytes) -> EthResult<H256> {
        let recovered = recover_raw_transaction(tx)?;

        let pool_transaction = <Pool::Transaction>::from_recovered_transaction(recovered);

        // forward the transaction to the optimism sequencer
        if self.network().chain_id() == 10 {
            let client = Client::new();

            let body = json![{
                "jsonrpc": "2.0",
                "method": "eth_sendRawTransaction",
                "params":[hex!(tx)],
                "id": 10,
            }];

            let req = Request::builder()
                .method(Method::POST)
                .uri("https://mainnet-sequencer.optimism.io")
                .header("Content-type", "application/json")
                .body(Body::from(body));

            let res = client.request(req).await?;
        }

        // submit the transaction to the pool with a `Local` origin
        let hash = self.pool().add_transaction(TransactionOrigin::Local, pool_transaction).await?;

        Ok(hash)
    }
```

**Comment by merklefruit** [comment]
This lgtm! Just as a reminder, this is a part of op-reth and won't live in this repo for now, so any changes should be opened as a PR on https://github.com/anton-rs/op-reth starting from the branch clabby/op-reth (which we use to see the entire diff so far).

Please don't be afraid to ask if you need any other info on contributing to this!

**Comment by merklefruit** [comment]
Closed by https://github.com/anton-rs/op-reth/pull/41
